applications/nrf_desktop/CMakeLists.txt

```diff
@@ -47,6 +47,62 @@ if(CONFIG_IMG_MANAGER)
   zephyr_library_link_libraries(MCUBOOT_BOOTUTIL)
 endif()
 
+if(CONFIG_BOOTLOADER_MCUBOOT)
+  if(NOT (CONFIG_MCUBOOT_BOOTLOADER_MODE_SINGLE_APP OR
+          CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP OR
+          CONFIG_MCUBOOT_BOOTLOADER_MODE_RAM_LOAD OR
+          CONFIG_MCUBOOT_BOOTLOADER_MODE_SWAP_USING_MOVE))
+    message(WARNING "
+            ------------------------------------------------------------------------------
+            --- WARNING: The currently selected mode of the MCUboot bootloader has not ---
+            --- been used or tested with any nRF Desktop configuration.                ---
+            ------------------------------------------------------------------------------
+            ")
+  endif()
+
+  if(CONFIG_MCUBOOT_BOOTLOADER_MODE_RAM_LOAD)
+    message(WARNING "
+            -----------------------------------------------------------------------------
+            --- WARNING: The RAM load mode of the MCUboot bootloader is experimental. ---
+            -----------------------------------------------------------------------------
+            ")
+
+    # The RAM load requires the code partition to be defined in DTS.
+    if(CONFIG_PARTITION_MANAGER_ENABLED OR (NOT CONFIG_USE_DT_CODE_PARTITION))
+      message(FATAL_ERROR
+              "The RAM load mode of the MCUboot bootloader is not supported with the Partition "
+              "Manager (PM). Please disable PM and use DTS for defining the memory layout.")
+    endif()
+
+    # Enforce the specific RAM layout in DTS when MCUboot uses the RAM load mode.
+    dt_nodelabel(app_rxm_region_node NODELABEL cpuapp_sram_app_rxm_region)
+    if(NOT app_rxm_region_node)
+      message(FATAL_ERROR
+              "Missing RXM (ROM + RAM) region definition (cpuapp_sram_app_rxm_region) for the "
+              "application image in DTS.")
+    endif()
+
+    dt_nodelabel(mcuboot_ram_region_node NODELABEL cpuapp_sram_mcuboot_ram_region)
+    if(NOT mcuboot_ram_region_node)
+      message(FATAL_ERROR
+              "Missing RAM region definition (cpuapp_sram_mcuboot_ram_region) for the MCUboot "
+              "bootloader in DTS.")
+    endif()
+
+    dt_reg_addr(app_rxm_region_addr PATH ${app_rxm_region_node})
+    dt_reg_addr(mcuboot_ram_region_addr PATH ${mcuboot_ram_region_node})
+    if(mcuboot_ram_region_addr LESS_EQUAL app_rxm_region_addr)
+      message(FATAL_ERROR
+              "The start address of the RXM region (0x${app_rxm_region_addr}) must be located "
```

Comment:
Consider mentioning that it's an app-specific assumption (not a generic one). Maybe something similar to:

`The validation process assumes a specific RAM layout in DTS - you can refer to the following file as an example: ./configuration/nrf54lm20dk_nrf54lm20a_cpuapp/memory_map_ram_load.dtsi`